packages/ts/react-signals/src/ValueSignal.ts

```diff
@@ -7,17 +8,41 @@ import {
 } from './events.js';
 import { $processServerResponse, $update, FullStackSignal } from './FullStackSignal.js';
 
-type PromiseWithResolvers = ReturnType<typeof Promise.withResolvers<void>>;
+/**
+ * The type of function that is used to define a callback that is called when
+ * the `then` method is called on an operation object.
+ */
+export type ThenCallback = () => void;
+
+/**
+ * A return type for signal operations.
+ */
+export type Operation = {
+  result: {
+    then(callback: ThenCallback): Operation['result'];
+  };
+};
+
+/**
+ * An operation where all callbacks are predefined to be no-ops.
+ */
+export const noOperation: Operation = Object.freeze({
+  result: {
+    then(_callback: ThenCallback): Operation['result'] {
+      return noOperation.result;
+    },
+  },
+});
```

Comment:
Could all this common definitions be moved to the `FullStackSignals.ts` so that other signals (e.g. ListSignal) can also leverage from the same mechanisms for making their operations `thenable`?

Reply:
Moved them, although I think that those kind of changed can be done when needed.